src/riverroute/MOSART_physics_mod.F90

```diff
@@ -1,710 +1,730 @@
+
+   subroutine Routing_KW(iunit, nt, theDeltaT)
+      !  classic kinematic wave routing method
+
+      ! Arguments
+      integer, intent(in) :: iunit, nt
+      real(r8), intent(in) :: theDeltaT
+      integer  :: k
+      real(r8) :: temp_gwl, temp_dwr, temp_gwl0
+
+      ! estimate the inflow from upstream units
+      TRunoff%erin(iunit,nt) = 0._r8
+
+      ! tcraig, moved this out of the inner main channel loop to before main channel call
+      ! now it's precomputed as TRunoff%eroutUp
+      !    do k=1,TUnit%nUp(iunit)
+      !       TRunoff%erin(iunit,nt) = TRunoff%erin(iunit,nt) - TRunoff%erout(TUnit%iUp(iunit,k),nt)
+      !    end do
+      TRunoff%erin(iunit,nt) = TRunoff%erin(iunit,nt) - TRunoff%eroutUp(iunit,nt)
+
+      ! estimate the outflow
+      if(TUnit%rlen(iunit) <= 0._r8) then ! no river network, no channel routing
+         TRunoff%vr(iunit,nt) = 0._r8
+         TRunoff%erout(iunit,nt) = -TRunoff%erin(iunit,nt)-TRunoff%erlateral(iunit,nt)
+      else
+         if(TUnit%areaTotal2(iunit)/TUnit%rwidth(iunit)/TUnit%rlen(iunit) > 1e6_r8) then
+            TRunoff%erout(iunit,nt) = -TRunoff%erin(iunit,nt)-TRunoff%erlateral(iunit,nt)
+         else
+            !        !TRunoff%vr(iunit,nt) = CRVRMAN(TUnit%rslp(iunit), TUnit%nr(iunit), TRunoff%rr(iunit,nt))
+            TRunoff%vr(iunit,nt) = CRVRMAN_nosqrt(TUnit%rslpsqrt(iunit), TUnit%nr(iunit), TRunoff%rr(iunit,nt))
+            TRunoff%erout(iunit,nt) = -TRunoff%vr(iunit,nt) * TRunoff%mr(iunit,nt)
+            if(-TRunoff%erout(iunit,nt) > TINYVALUE .and. TRunoff%wr(iunit,nt) + &
+              (TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%erout(iunit,nt)) * theDeltaT < TINYVALUE) then
+               TRunoff%erout(iunit,nt) = &
+                    -(TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%wr(iunit,nt) / theDeltaT)
+               if(TRunoff%mr(iunit,nt) > 0._r8) then
+                  TRunoff%vr(iunit,nt) = -TRunoff%erout(iunit,nt) / TRunoff%mr(iunit,nt)
+               end if
+            end if
+         end if
+      end if
+
+      temp_gwl = TRunoff%qgwl(iunit,nt) * TUnit%area(iunit) * TUnit%frac(iunit)
+
+      TRunoff%dwr(iunit,nt) = TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%erout(iunit,nt) + temp_gwl
+
+      if((TRunoff%wr(iunit,nt)/theDeltaT &
+           + TRunoff%dwr(iunit,nt)) < -TINYVALUE) then
+         write(iulog,*) 'mosart: ERROR main channel going negative: ', iunit, nt
+         write(iulog,*) theDeltaT, TRunoff%wr(iunit,nt), &
+              TRunoff%wr(iunit,nt)/theDeltaT, TRunoff%dwr(iunit,nt), temp_gwl
+         write(iulog,*) ' '
+         !       call shr_sys_abort('mosart: ERROR main channel going negative')
+      endif
+
+      ! check for stability
+      !    if(TRunoff%vr(iunit,nt) < -TINYVALUE .or. TRunoff%vr(iunit,nt) > 30) then
+      !       write(iulog,*) "Numerical error inRouting_KW, ", iunit,nt,TRunoff%vr(iunit,nt)
+      !    end if
```

Comment:
Should we just remove the commented code?

Reply:
@yes - I think that would be great. I'll do that.